modules/tinymce/src/core/main/ts/util/ArrUtils.ts

```diff
@@ -5,33 +5,40 @@
  * For commercial licenses see https://www.tiny.cloud/
  */
 
+import { Type } from '@ephox/katamari';
+
 /**
  * Array utility class.
  *
  * @private
  * @class tinymce.util.Arr
  */
 
-const isArray = Array.isArray;
+export type ArrayCallback<T, R> = (x: T, i: number, xs: ReadonlyArray<T>) => R;
+export type ObjCallback<T, R> = (value: T, key: string, obj: Record<string, T>) => R;
 
-const toArray = function (obj) {
-  let array = obj, i, l;
+const isArray = Array.isArray;
 
+const toArray = <T>(obj: ArrayLike<T>): T[] => {
   if (!isArray(obj)) {
-    array = [];
-    for (i = 0, l = obj.length; i < l; i++) {
+    const array: T[] = [];
+    for (let i = 0, l = obj.length; i < l; i++) {
       array[i] = obj[i];
     }
+    return array;
+  } else {
+    return obj;
   }
-
-  return array;
 };
 
-const each = function (o, cb, s?) {
+const each: {
+  <T>(arr: ArrayLike<T> | null | undefined, cb: ArrayCallback<T, void | boolean>, scope?: any): boolean;
+  <T>(obj: Record<string, T> | null | undefined, cb: ObjCallback<T, void | boolean>, scope?: any): boolean;
+} = (o, cb, s?): boolean => {
   let n, l;
 
   if (!o) {
-    return 0;
+    return false;
```

Comment:
Oh no, what if I get nostalgic for C-style booleans in my typescript codebase though? :smiling_imp:

Reply:
Yeah, this is actually a functionality change, however the intention here clearly is to be using booleans.